### PR TITLE
perf(nuxt): use faster approach to check cache exists

### PR DESCRIPTION
**Pull request by Mini-ghost**
### 🔗 Linked issue

N/A

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Currently, the implementation of `hasCachedData` inside `useAsyncData` is as follows:

```ts
const hasCachedData = () => ![null, undefined].includes(options.getCachedData!(key) as any)
```

Perhaps we can tune it for better performance by simplifying the condition like this:

```ts
const hasCachedData = () => options.getCachedData!(key) != null
```

For reference, here are the benchmark test results:

```
check via `value != null` x 993,997,209 ops/sec ±0.10% (98 runs sampled)
check via `![null, undefined].includes(value)` x 74,629,930 ops/sec ±0.35% (98 runs sampled)
Fastest is check via `value != null`
```

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by danielroe — APPROVED** [review]

**Review by TheAlexLichter — APPROVED** [review]

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]
What if `0`, `false` or the like are returned? 🤔

File: packages/nuxt/src/app/composables/asyncData.ts [hunk 1]
```diff
@@ -243,7 +243,7 @@ export function useAsyncData<
     console.warn('[nuxt] `boolean` values are deprecated for the `dedupe` option of `useAsyncData` and will be removed in the future. Use \'cancel\' or \'defer\' instead.')
   }
 
-  const hasCachedData = () => ![null, undefined].includes(options.getCachedData!(key) as any)
+  const hasCachedData = () => options.getCachedData!(key) != null
```
Comment (manniL) [inline packages/nuxt/src/app/composables/asyncData.ts:246]:
```suggestion
  const hasCachedData = () => {
    const data = options.getCachedData!(key)
    return data !== null && data !== undefined
 }
```

Maybe explicitly check null and undefined instead?

Comment (Mini-ghost) [reply packages/nuxt/src/app/composables/asyncData.ts:246]: In fact, `value != null` is used in some places inside Nuxt. I think this perhaps clear enough to identify `undefined` or `null` 🤔

Comment (manniL) [reply packages/nuxt/src/app/composables/asyncData.ts:246]: But what if the cached data should return `false`, `0` or `''`? 🤔

Comment (Mini-ghost) [reply packages/nuxt/src/app/composables/asyncData.ts:246]: However, it's worth noting that encountering a `null` or `undefined` return value will trigger a fetch, as clearly outlined in the documentation. May I suggest that adhering to this established behavior might be the preferable approach?

https://github.com/nuxt/nuxt/blob/8bb8d824c73a7b2e4fcc4f0b870e6e036318e6a6/packages/nuxt/src/app/composables/asyncData.ts#L59-L64

Comment (danielroe) [reply packages/nuxt/src/app/composables/asyncData.ts:246]: @manniL I also had to check, but I think it's good:

CleanShot 2024-03-10 at 10 23 02@2x

Comment (manniL) [reply packages/nuxt/src/app/composables/asyncData.ts:246]: 🤦  I think I accidentally checked for `!= undefined` in my tests.

 LGTM!

**Review by TheAlexLichter — APPROVED** [review]

**Pull request merged**